codec/reflectcodec/type_codec.go

```diff
@@ -332,6 +349,70 @@ func (c *genericCodec) marshal(value reflect.Value, p *wrappers.Packer, maxSlice
 				return err
 			}
 		}
+		return nil
+	case reflect.Map:
+		keys := value.MapKeys()
+		numElts := len(keys)
+		if uint32(numElts) > maxSliceLen {
+			return fmt.Errorf("%w; slice length, %d, exceeds maximum length, %d",
+				codec.ErrMaxSliceLenExceeded,
+				numElts,
+				maxSliceLen,
+			)
+		}
+		p.PackInt(uint32(numElts)) // pack # elements
+		if p.Err != nil {
+			return p.Err
+		}
+
+		type keyTuple struct {
+			key    reflect.Value
+			binary []byte
+		}
+
+		sortedKeys := make([]keyTuple, 0)
+
+		for _, key := range keys {
+			p := wrappers.Packer{
+				MaxSize: int(c.maxSliceLen),
+				Bytes:   make([]byte, 0, 128),
+			}
+			if err := c.marshal(key, &p, c.maxSliceLen); err != nil {
+				return err
+			}
+			sortedKeys = append(sortedKeys, keyTuple{
+				key:    key,
+				binary: p.Bytes,
+			})
+		}
```

Comment:
Assignment is a bit faster than appending, and this is a hot path so it's worth it

```suggestion
		sortedKeys := make([]keyTuple, len(keys))

		for i, key := range keys {
			p := wrappers.Packer{
				MaxSize: int(c.maxSliceLen),
				Bytes:   make([]byte, 0, 128),
			}
			if err := c.marshal(key, &p, c.maxSliceLen); err != nil {
				return err
			}
			sortedKeys[i] = keyTuple{
				key:    key,
				binary: p.Bytes,
			}
		}
```

Reply:
Thank you!